src/test/java/de/fraunhofer/aisec/cpg/enhancements/variable_resolution/VariableResolverJavaTest.java

```diff
@@ -0,0 +1,306 @@
+
+// Todo VariableResolver 7 Failed, 7 Passed
+
+@Disabled(
+    "Until changing variable resolution to ScopeManager. Then in detail disable the tests that need specific fixes")
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class VariableResolverJavaTest extends BaseTest {
+
+  // Externally defined static global
+
+  private RecordDeclaration externalClass;
+  private FieldDeclaration externVarName;
+  private FieldDeclaration externStaticVarName;
+
+  private RecordDeclaration outerClass;
+  private FieldDeclaration outerVarName;
+  private FieldDeclaration outerStaticVarName;
+  private FieldDeclaration outerImpThis;
+
+  private RecordDeclaration innerClass;
+  private FieldDeclaration innerVarName;
+  private FieldDeclaration innerStaticVarName;
+  private FieldDeclaration innerImpThis;
+  private FieldDeclaration innerImpOuter;
+
+  private MethodDeclaration main;
+
+  private MethodDeclaration outerFunction1;
+  private List<ForStatement> forStatements;
+  private MethodDeclaration outerFunction2;
+  private MethodDeclaration outerFunction3;
+  private MethodDeclaration outerFunction4;
+
+  private MethodDeclaration innerFunction1;
+  private MethodDeclaration innerFunction2;
+  private MethodDeclaration innerFunction3;
+
+  private Map<String, Expression> callParamMap = new HashMap<>();
+
+  @BeforeAll
+  public void initTests() throws ExecutionException, InterruptedException {
+    final String topLevelPath = "src/test/resources/variables_extended/";
+    List<String> fileNames = Arrays.asList("ScopeVariables.java", "ExternalClass.java");
+    List<File> fileLocations =
+        fileNames.stream()
+            .map(fileName -> new File(topLevelPath + fileName))
+            .collect(Collectors.toList());
+    TranslationConfiguration config =
+        TranslationConfiguration.builder()
+            .sourceLocations(fileLocations.toArray(new File[fileNames.size()]))
+            .topLevel(new File(topLevelPath))
+            .defaultPasses()
+            .debugParser(true)
+            .failOnError(true)
+            .build();
+
+    TranslationManager analyzer = TranslationManager.builder().config(config).build();
+    List<TranslationUnitDeclaration> tu = analyzer.analyze().get().getTranslationUnits();
+
+    List<Node> nodes =
+        tu.stream()
+            .flatMap(tUnit -> SubgraphWalker.flattenAST(tUnit).stream())
+            .collect(Collectors.toList());
+    List<CallExpression> calls =
+        Util.filterCast(nodes, CallExpression.class).stream()
+            .filter(call -> call.getName().equals("printLog"))
+            .collect(Collectors.toList());
+    calls.sort(new NodeComparator());
+
+    List<RecordDeclaration> records = Util.filterCast(nodes, RecordDeclaration.class);
+
+    // Extract all Variable declarations and field declarations for matching
+    externalClass =
+        Util.getOfTypeWithName(nodes, RecordDeclaration.class, "variables_extended.ExternalClass");
```

Comment:
Using the workflow mentioned above would also make constructs like this much less verbose

Reply:
This would then be e.g.
```java
List<RecordDeclaration> records = Util.subnodesOfType(tu, RecordDeclaration.class);
externalClass = TestUtils.findByUniqueName(records, "variables_extended.ExternalClass");
```

Reply:
> verbose

Do you mean verbose or readable, are there logs written when using these?

Reply:
I mean verbose in the sense of a lot of code

Reply:
I found two places where there is a `filterCast` and then a filter by name, there I decided to adopt the proposal because it is only two places and an additional function would not have been worth the reduction in code duplication.  nested `filterCast` and `findByName`